### PR TITLE
Dev/adis16475 tests

### PR DESCRIPTION
**Pull request by rbolboac**
# Description
Add new tests for adis16475 driver and change the xml to the latest version of the drivers to include also the delta channels.
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A
- [x] Test B

Test Configuration: A

    Hardware:ADIS16505-2 with RPI4
    OS:Kuiper with ADI Kernel 6.1.54

Test Configuration: B

    Hardware:ADIS16505-2 with SDPK1
    OS:No-OS master branch

# Documentation

If this is a new feature or example please mention or link any documentation. All new hardware interface classes require documentation.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by rbolboac** [comment]
Hello @tfcollins ,
Any remarks here?
Thank you!

File: test/test_adis16475_p.py [hunk 1]
```diff
@@ -1,15 +1,121 @@
...
+
+@pytest.mark.iio_hardware(hardware, True)
+def test_adis16475_sample_rate(iio_uri):
+    adis16475 = adi.adis16475(uri=iio_uri)
+
+    for i in range(1, 2000):
+        adis16475.sample_rate = 2000.0 / (i + 1)
+        assert adis16475.sample_rate >= (2000.0 / (i + 1) - 1)
+        assert adis16475.sample_rate <= (2000.0 / (i + 1) + 1)
+    adis16475.sample_rate = 2000.0
```
Comment (tfcollins) [inline test/test_adis16475_p.py:60]: Is this needed for downstream tests? If so either use a fixture to create a test dependency

File: adi/adis16475.py [hunk 2]
```diff
@@ -65,6 +65,19 @@ def __init__(self, uri="", device_name="adis16505-2"):
...
+                + device_name
+                + ". Searching for a device found in the compatible list."
+            )
+            for i in compatible_parts:
+                self._ctrl = self._ctx.find_device(i)
+                self._rxadc = self._ctx.find_device(i)
+                if self._ctrl is not None:
+                    print("Fond device = " + i + ". Will use this device instead.")
+                    break
+
```
Comment (tfcollins) [inline adi/adis16475.py:80]: Need a check to make sure a device is found

**Review by tfcollins — CHANGES_REQUESTED** [review]

File: adi/adis16475.py (hunk 2 above)
Comment (rbolboac) [reply adi/adis16475.py:82]: I added an exception in case no device is found

**Comment by rbolboac** [comment]
Hello @tfcollins ,

Any feedback after new changes?

Thank you!

**Pull request merged**